spec.cue

```diff
@@ -0,0 +1,61 @@
+package spec
+
+version: 2
+sources: [...#Source]
+
+
+#StringInt: string | number
+
+#Partition: {
+	name:        string & !=""
+	data_type:   string & !=""
+	expression?: string
+	vals?: {
+		macro: string & !=""
+		args: [string]: #StringInt
+	}
+	path_macro?: string & !=""
+}
+
+#SourceSnowflake: {
+	location:     string & !=""
+	file_format:  string & !=""
+	auto_refresh: bool
+	partitions: [...#Partition]
+}
+
+#SourceRedshift: {
+	location:   string & !=""
+	row_format: string & !=""
+	partitions: [...#Partition]
+}
+
+#SourceSpark: {
+	location: string & !=""
+	using:    string & !=""
+	options: [string]: string
+}
+
+#Column: {
+	name:         string & !=""
+	data_type:    string & !=""
+	description?: string
+}
+
+#Table: {
+	name:     string & !=""
+	external: #SourceRedshift | #SourceSnowflake | #SourceSpark
+	table_properties: [string]: string
+	columns: [...#Column]
+}
+
+#Loaders: "S3"
```

Comment:
I looked in the repo to see which loaders were supported an only found references to `S3`. Are any other loaders supported? 

If so this statement would become:

```
#Loaders: "S3" | "AzureBlob" | "Etc"
```

Reply:
this can be any string ([docs](https://docs.getdbt.com/reference/resource-properties/loader/))